src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -2534,6 +2540,19 @@ void ReadParameterStatus(ReadOnlySpan<byte> incomingName, ReadOnlySpan<byte> inc
             return null;
         }
 
+        internal void CommandStart(string sql)
+        {
+            Debug.Assert(_currentActivity is null);
+            if (NpgsqlActivitySource.IsEnabled)
+                _currentActivity = NpgsqlActivitySource.CommandStart(this, sql);
+        }
+
+        internal void StopCurrentActivity()
+        {
+            _currentActivity?.Dispose();
+            _currentActivity = null;
```

Comment:
Perf-wise we can avoid setting this to null - no real need right?

Reply:
That depends - if we do move `StopCurrentActivity` to `EndUserAction`, this potentially might lead to double-dispose.

Reply:
Since it already checks for null for dispose, we might just as well check for both (by adding `if (_currentActivity is not null)`).

Reply:
Am OK with whatever here